### PR TITLE
Fix is_multiple_of to account for 0

### PR DESCRIPTION
**Pull request by WizardOfMenlo**
This pull request fixes the implementation to `is_multiple_of` to account for the case in which one of the operand is zero. The current implementation panics, while this one currently concludes that 0 is a multiple of 0 (and in fact the only one).

The change is a minimal check, and an inclusion of this edge case in the tests.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -981,6 +987,7 @@ macro_rules! impl_integer_for_usize {
 
             #[test]
             fn test_is_multiple_of() {
+                assert!((0 as $T).is_multiple_of(&(0 as $T)));
                 assert!((6 as $T).is_multiple_of(&(6 as $T)));
                 assert!((6 as $T).is_multiple_of(&(3 as $T)));
                 assert!((6 as $T).is_multiple_of(&(1 as $T)));
```
Comment (cuviper) [inline src/lib.rs:993]: Hmm, these tests have no negatives -- we could pass with `return true`. Could I get you to add a few variants, including a false test with RHS 0?

Comment (WizardOfMenlo) [reply src/lib.rs:993]: Done!

**Comment by cuviper** [comment]
Thanks!

bors r+
